ansible/playbooks/roles/repository/files/download-requirements/src/mode/red_hat_family_mode.py

```diff
@@ -87,14 +87,26 @@ def _add_third_party_repositories(self):
 
     def __remove_dnf_cache_for_untracked_repos(self):
         # clean metadata for upgrades (when the same package can be downloaded from changed repo)
-        repoinfo: List[str] = self._tools.dnf.list_all_repos_info()
-        repocaches: List[str] = [repodir for repodir in Path('/var/cache/dnf').iterdir() if repodir.is_dir()]
-
-        for repo in repoinfo:
-            if repo['Repo-status'] == 'disabled':
-                for repocache in repocaches:
-                    if repocache.name.startswith(repo['Repo-id']):
+        repocaches: List[str] = list(Path('/var/cache/dnf').iterdir())
```

Comment:
The path `/var/cache/dnf` should not be hardcoded, this is the default value but it can be changed via `cachedir` property in `/etc/dnf/dnf.conf`.